src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,655 @@ public ByteArraySet createSmallSet() {
         return set;
     }
 
+    public List<ByteArray> getRandomByteArrays(int count, long seed) {
+        System.out.println("Generating random " + count + " Byte Array(s) with " + seed + " seed");
+        Random randomSeed = new Random(seed);
+        List<ByteArray> vals = Stream.generate(() -> RandomStringUtils.random(RANDOM_STRING_SIZE, 0, 0, true, true, null, randomSeed))
+                .limit(count)
+                .collect(Collectors.toSet())
+                .stream()
+                .map(str -> new ByteArray(str.getBytes()))
+                .collect(Collectors.toList());
+        assertEquals(count, vals.size());
+        return vals;
+    }
+
+    public List<ByteArray> getRandomByteArrays(int count) {
+        return getRandomByteArrays(count, RANDOM_SEED);
+    }
+
+    public void testAdd(int size) {
+        System.out.println("Testing add method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        List<ByteArray> insertedVals = new ArrayList<ByteArray>();
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+            assertTrue(set.contains(val));
+            insertedVals.add(val);
+            assertEquals(insertedVals.size(), set.size());
+        }
+
+        for (ByteArray val: vals) {
+            assertFalse(set.add(val));
+            assertTrue(set.contains(val));
+        }
+        assertEquals(size, set.size());
+
+        for (ByteArray val: vals) {
```

Comment:
This loop looks redundant to the loop above. You already iterate through `vals` once and call `set.contains()` on all after calling `set.add`